src/components/Table/index.tsx

```diff
@@ -0,0 +1,123 @@
+import { ColumnDef, flexRender, getCoreRowModel, useReactTable } from '@tanstack/react-table'
+import { useLayoutEffect, useRef, useState } from 'react'
+import styled from 'styled-components'
+
+const Wrapper = styled.div`
+  border: 1px solid ${({ theme }) => theme.surface3};
+  border-radius: 20px;
+  overflow: hidden;
+`
+const TableContainer = styled.div`
+  overflow: auto; // Enable horizontal scroll on full table
+`
+const TableHeadContainer = styled.div`
+  // Enable horizontal scroll on the header but hide scrollbar
+  overflow-x: auto;
+  ::-webkit-scrollbar {
+    display: none;
+  }
+  -ms-overflow-style: none;
+  scrollbar-width: none;
+`
+const TableBodyContainer = styled.div`
+  max-height: 550px;
+  overflow-y: auto; // Enable vertical scroll only on table body
+`
+const TableSection = styled.table`
+  border-collapse: collapse;
+  box-sizing: border-box;
+`
+const TableHead = styled.thead`
+  display: table;
+  width: 100%;
+`
+const TableBody = styled.tbody`
+  display: table;
+  width: 100%;
+`
+const TableRow = styled.tr`
+  :hover {
+    background: ${({ theme }) => theme.surface3};
+  }
+`
+export const TableCell = styled.div<{ alignRight?: boolean }>`
+  width: 100%;
+  display: flex;
+  padding: 12px 16px;
+  white-space: nowrap;
+  justify-content: ${({ alignRight }) => (alignRight ? 'flex-end' : 'flex-start')};
+`
+const TableHeader = styled.th<{ width: number }>`
+  min-width: ${({ width }) => width}px;
+  padding: 8px;
+  border-bottom: 1px solid ${({ theme }) => theme.surface3};
+`
+const TableData = styled.td<{ width: number }>`
```

Comment:
all inputs that share the name of a style should be prefixed with `$` ie `$width` and `$alignRight`